### PR TITLE
refactor: Create structs for each command to  better isolate commands

### PR DESCRIPTION
**Pull request by soltysh**
## Description
This refactor is a first step to isolating all the command, which should allow easier consumption of zarf in other places. 

## Related Issue
Relates to #2773

## Checklist before merging

- [x] Test, docs, adr added or updated as needed
- [x] [Contributor Guide Steps](https://github.com/zarf-dev/zarf/blob/main/CONTRIBUTING.md#developer-workflow) followed

### CODE REVIEW

File: src/cmd/dev.go [hunk 1]
```diff
@@ -31,378 +31,465 @@ import (
...
+		Short:   lang.CmdDevPatchGitShort,
+		Args:    cobra.ExactArgs(2),
+		RunE:    o.Run,
+	}
 
-		if helpers.IsURL(fileName) {
-			message.Warn(lang.CmdDevSha256sumRemoteWarning)
-			logger.From(cmd.Context()).Warn("this is a remote source. If a published checksum is available you should use that rather than calculating it directly from the remote link")
+	// TODO(soltysh): get rid of pkgConfig global
+	cmd.Flags().StringVar(&pkgConfig.InitOpts.GitServer.PushUsername, "git-account", types.ZarfGitPushUser, lang.CmdDevFlagGitAccount)
```
Comment (AustinAbro321) [inline src/cmd/dev.go:191]: Yup the globals are moving away in #2969

File: src/cmd/connect.go [hunk 2]
```diff
@@ -16,100 +16,121 @@ import (
...
 }
 
-var connectListCmd = &cobra.Command{
-	Use:     "list",
-	Aliases: []string{"l"},
-	Short:   lang.CmdConnectListShort,
-	RunE: func(cmd *cobra.Command, _ []string) error {
-		c, err := cluster.NewCluster()
+// Run performs the execution of 'connect' sub command.
+func (o *ConnectOptions) Run(cmd *cobra.Command, args []string) error {
```
Comment (mkcp) [inline src/cmd/connect.go:51]: Good choice to isolate these out into function declarations on the pkg

File: src/cmd/connect.go [hunk 3]
```diff
@@ -16,100 +16,121 @@ import (
...
+	l := logger.From(ctx)
+	target := ""
+	if len(args) > 0 {
+		target = args[0]
+	}
+
+	spinner := message.NewProgressSpinner(lang.CmdConnectPreparingTunnel, target)
+	defer spinner.Stop()
+
+	c, err := cluster.NewCluster()
```
Comment (mkcp) [inline src/cmd/connect.go:62]: Definitely not in scope for this PR, but sometime in the future it'd be great to ensure every CLI command can call a clearly defined API fn that encapsulates all of the configuration. Connect in particular has a lot of logic that could be abstracted out of the cli layer, which made me think of this.

File: src/cmd/connect.go [hunk 4]
```diff
@@ -16,100 +16,121 @@ import (
...
+	case err = <-tunnel.ErrChan():
+		return fmt.Errorf("lost connection to the service: %w", err)
+	}
 }
 
-func init() {
-	rootCmd.AddCommand(connectCmd)
-	connectCmd.AddCommand(connectListCmd)
+// ConnectOptions holds the command-line options for 'connect list' sub-command.
+type ConnectListOptions struct{}
```
Comment (mkcp) [inline src/cmd/connect.go:110]: +1 good pattern

File: src/cmd/root.go [hunk 5]
```diff
@@ -126,6 +116,31 @@ func run(cmd *cobra.Command, _ []string) {
...
+		Run:               run,
+	}
+
+	// TODO(soltysh): consider adding command groups
+	rootCmd.AddCommand(NewConnectCommand())
+	rootCmd.AddCommand(NewDestroyCommand())
+	rootCmd.AddCommand(NewDevCommand())
+	rootCmd.AddCommand(NewInitCommand())
+	rootCmd.AddCommand(NewInternalCommand(rootCmd))
+	rootCmd.AddCommand(NewPackageCommand())
```
Comment (mkcp) [inline src/cmd/root.go:139]: This is the biggest win of the refactor imo -- we don't need to understand what magic implicit invocation order you get from a bunch of init()s being executed. Clear and declarative 👍 

One note though, `zarf say`'s command should probably be added to this sequence. It's currently in Execute below, but it's better off grouped with everything else.

**Review by mkcp — COMMENTED** [review]
This is a good direction for cmd imo. Lots of room for future improvements as well.

File: src/cmd/root.go (hunk 5 above)
Comment (soltysh) [reply src/cmd/root.go:139]: Yes!

The other part I actually started considering is grouping commands, have a look at `kubectl` output, it groups commands into several groups (basic, deploy, cluster management, etc...). We don't have that many, yet, but if we want to consider this construct will make it very easy to do just that.

Comment (mkcp) [reply src/cmd/root.go:139]: That makes a lot of sense, i'm in favor!

**Review by AustinAbro321 — APPROVED** [review]
LGTM

**Pull request merged**